src/pid.cpp

```diff
@@ -335,21 +335,16 @@ double Pid::computeCommand(double error, double error_dot, ros::Duration dt)
   // Calculate the integral of the position error
   i_error_ += dt.toSec() * p_error_;
 
-  if(gains.antiwindup_)
-  {
-    // Prevent i_error_ from climbing higher than permitted by i_max_/i_min_
-    i_error_ = boost::algorithm::clamp(i_error_,
-                                       gains.i_min_ / std::abs(gains.i_gain_),
-                                       gains.i_max_ / std::abs(gains.i_gain_));
-  }
-
   // Calculate integral contribution to command
   i_term = gains.i_gain_ * i_error_;
 
-  if(!gains.antiwindup_)
+  // Limit i_term so that the limit is meaningful in the output
+  i_term = boost::algorithm::clamp(i_term, gains.i_min_, gains.i_max_);
+
+  if(gains.antiwindup_ && gains.i_gain_!=0)
   {
-    // Limit i_term so that the limit is meaningful in the output
-    i_term = boost::algorithm::clamp(i_term, gains.i_min_, gains.i_max_);
+    // Prevent i_error_ from climbing higher than permitted by i_max_/i_min_
+    i_error_ = i_term / gains.i_gain_;
```

Comment:
this does change the value every time..

I'd like to use the fixed clamping instead (not yet tested):
```c++
std::pair<double> bounds = std::minmax(gains.i_min_ / gains.i_gain_, gains.i_max_);
i_error_ = boost::algorithm::clamp(i_error_, bounds.first, bound.second);	
```
And please place it at the old location, this way there is no disparity

Reply:
That seems great, but `std::minmax` may be a C++11 function. I can't use it unless I change CMakeLists.txt. So I used `std::min` and `std::max`.